conans/client/installer.py

```diff
@@ -595,6 +595,8 @@ def _call_package_info(self, conanfile, package_folder, ref):
                     self._hook_manager.execute("pre_package_info", conanfile=conanfile,
                                                reference=ref)
                     conanfile.package_info()
+                    for comp_name in conanfile.cpp_info.components:
+                        conanfile.cpp_info.components[comp_name].version = conanfile.cpp_info.version
```

Comment:
It would be better to have the version assigned by default in the model (if possible), the same way we are already propagating the root path value here:

https://github.com/conan-io/conan/blob/17c8729e16db27c0e9352bb641c43ecac4b9d6c5/conans/model/build_info.py#L199